src/messages/MessageElement.hpp

```diff
@@ -321,6 +319,38 @@ class EmoteElement : public MessageElement
     EmotePtr emote_;
 };
 
+class LayeredEmoteElement : public MessageElement
```

Comment:
Could you add a class comment for this? It should include what purpose it has/what it tries to solve, not necessarily any technical details for how it solves it.